### ISSUE TITLE
Fallback to db-single flag with default disable

### ISSUE THREAD

**Issue by shreddedbacon** [uselagoon/build-deploy-tool#56]
I think we should disable the fallback to `xdb-single` by default, and have a flag to enable the fallback feature in
* .lagoon.yml
  * `allowDatabaseFallback: true|false` ?
* docker-compose service label
  * `lagoon.databasefallback: true|false`
* lagoon api environment variable
  * `LAGOON_FEATURE_FLAG_DATABASE_FALLBACK: enabled|disabled|true|false`

This would mean that builds will FAIL if there is no dbaas-provider present to take the workload, which seems desirable as it would prevent builds from creating db pods and then trying to figure out what went wrong and fixing the environment to use a dbaas provider.

**Comment by shreddedbacon** [comment]
With work I'm doing on the dbaas-provider configuration in this tool, and looking at past experiences in platform with builds that fail to provision against a dbaas for any number of reasons (requested environment type doesn't have a provider, the dbaas-operator is not responding/not installed), a build failure would have been much better and caused less issues for fixing it after the fact.

I know that we do have some users that WANT db-single pods, and they've already accounted for these in their docker-compose files or lagoon.yml as overrides, this behaviour wouldn't change.

**Comment by seanhamlin** [comment]
Perhaps fail should be the default, and only if there is some positive action by the user (e.g. what you have above) then it should continue with a pod. Agree, this for larger clusters is a pain.

**Comment by shreddedbacon** [comment]
Yep, I'm proposing failure is the default

**Comment by shreddedbacon** [comment]
@tobybellwood would be good to get some thoughts on this, so I can potentially roll this into #115

**Comment by shreddedbacon** [comment]
bump @tobybellwood

**Comment by tobybellwood** [comment]
I agree that a fallback default is a bad thing:tm: , and should only really be enabled at a cluster level (to account for a missing dbaas). If a project, or branch, wants a single - they can define it in their docker-compose - I don't think we need a complex triple override system?
